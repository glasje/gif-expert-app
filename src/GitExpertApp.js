import React, { useState } from 'react'
import PropTypes from 'prop-types'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GitExpertApp =({defaultCategories = [ ]})=> {
    
    //const categories = ['One Punch','Samurai X','Dragon Ball Z'];
    const [categories, setCategories] = useState(defaultCategories);
    
    return (
        <div>
            <h1>GitExpertApp</h1> 
           
            <AddCategory
                setCategories={setCategories}
            />
           <hr/>
            <ol>
                {categories.map((category,i) =>(
                    <GifGrid 
                    key={i}
                    category={category}/>
                ))}
            </ol>
        </div>
    )
}

GitExpertApp.propTypes = {

}

export default GitExpertApp;

