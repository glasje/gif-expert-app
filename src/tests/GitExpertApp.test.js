import { shallow } from 'enzyme';
import React from 'react';
import {GitExpertApp} from '../GitExpertApp'
describe('Pruebas en <GitExpertApp/>', () => {
    test('debe de mostrar correctamente', () => {
        const wrapper = shallow(<GitExpertApp />)
        expect(wrapper).toMatchSnapshot()
    })

    test('debe de mostrar una categoria de elementos', () => {
        const categories = ['One Punch','Dragon Ball Z'];
        const wrapper = shallow(<GitExpertApp defaultCategories={categories} />)
        expect(wrapper.find('GifGrid').length).toBe(categories.length)

    })
    
})
