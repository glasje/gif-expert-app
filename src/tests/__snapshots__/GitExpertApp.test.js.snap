// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GitExpertApp/> debe de mostrar correctamente 1`] = `
<div>
  <h1>
    GitExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</div>
`;
